### PR TITLE
max price calculation - construction price index - before 2011

### PR DESCRIPTION
**Pull request by toikarin-vincit**
# Hitas Pull Request

# Description

7720fb8: docker-compose: update swagger-editor

---

4994a3c: backend: update dependencies

---

33e4f97: frontend: update dependencies

---

07eb7a8: backend: oracle migration: fix miration without --skip-anonymize

---

41d15ca: backend: improve months_between_dates calculation
 - take days into account, don't just assume months
 - added tests

---

452c742: backend: implement depreciation percentage
 - We haven't found a correct formula for this yet and haven't had
   enough time to check for it. Added a simple implementation
   that helps us going forward for now but it's far from perfect.

---

2b444c9: backend: add temporary workaround for marking apartment calculation
 - it's now possible to add a note `FIXME: old rules` to an
   apartment and it forces to use old hitas rules
 - this is temporary solution until the proper way is implemented

---

611bb6c: backend: max price calculation with construction price index pre 2011
 - implement max price calculation with construction price index
   for apartments completed before 2011
 - improvement handling changes a lot
   - refactored improvement rules for each "case"
     - rules 2011 onwards
     - rules before 2011 - construction price index
     - rules before 2011 - market price index
 - API responses changes drastically
   - return different kind of calculation variables object
     for each "case"
     - apartments completed 2011 onwards
     - apartments completed before 2011 - construction price index
       - housing company improvements has its own type
       - apartment improvements has its own type
     - apartments completed before 2011 - market price index
       - housing company improvements has its own type
       - apartment improvements has its own type
 - all the rules and openapi types are fully decoupled for now
   - this is an (at least) the first initial step towards
     making it easier to read the different rulesets and
     how everything is different. This is probably an
     overkill but can be adjusted in future commits

---

613967b: backend: add new field 'new_hitas' to maximum price calculations
 - makes it a bit easier to see which kinds of response calculation
   variables are excected to be returned. Old hitas rules are
   not always only used for apartments completed before 2011,
   there are exceptions

---

1196ff2: backend: openapi: added examples
 - added two maximum price calculation examples
   - one for apartment completed 2011-01 onwards
   - one for apartment completed before 2011-01

---

b9554f7: backend: openapi: fix few errors in examples
 - wrong field name in bad request example
 - wrong enum value in not acceptable example

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [x] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [x] Automatic tests has been added
    - [x] OpenAPI definitions have been updated
    - [ ] Database migrations will work in test environment
    - [x] Oracle migration has been updated

## Test plan

Make a bunch of maximum price calculations for different apartments (before 2011, after)

## Tickets

This pull request resolves all or part of the following ticket(s): HT-37 HT-55 HT-80

### CODE REVIEW

File: backend/hitas/calculations/max_prices/max_price.py [hunk 1]
```diff
@@ -157,6 +159,7 @@ def calculate_max_price(
         "valid_until": valid_until,
         "maximum_price": max_price,
         "index": max_index,
+        "new_hitas": new_hitas_rules,
```
Comment (ranta) [inline backend/hitas/calculations/max_prices/max_price.py:162]: I prefer using more explicit variable names, e.g.: `use_new_hitas_rules` :)

**Review by ranta — APPROVED** [review]
🐸

File: backend/hitas/calculations/depreciation_percentage.py [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+# |   0-6 years |    1-72 months | always 1                                               | yes       |
+# |  6-11 years |  73-133 months | ROUND(0.99979165^(<months>-72), 4)                     | yes       |
+# | 11-16 years | 134-192 months | ROUND(<133 month value> * 0.9995825^(<months>-133), 4) | no        |
+# | 16+ years   | 193+ months    | ROUND(<192 month value> * 0.9991665^(<months>-192), 4) | no        |
+# +-------------+----------------+--------------------------------------------------------+-----------|
+#
+# It's hard to say if the values here are absolutely correct or not as these are hand-written to the old system and
+# there might be mistakes in these values too.
+
+_values = [
```
[the remark this answers is not in the document]
Comment (toikarin-vincit) [reply backend/hitas/calculations/depreciation_percentage.py]: fixed

File: backend/hitas/calculations/max_prices/max_price.py (hunk 1 above)
Comment (toikarin-vincit) [reply backend/hitas/calculations/max_prices/max_price.py:162]: I opted to use `new_hitas` as it's consistent with our filter parameter with a same name :)

**Pull request merged**